ComSemApp/teacher/views.py

```diff
@@ -355,3 +357,93 @@ def delete_file(url):
+
+class SpeakingPracticeReviewRequestsListView(TeacherCourseViewMixin, ListView):
+    model = SpeakingPracticeAttemptReviewRequest
+    template_name = 'ComSemApp/teacher/request_list.html'
+    context_object_name = 'requests'
+
+    def get_queryset(self):
+        requests_queryset = self.course.get_speaking_practice_review_requests().order_by('review','-date')
+
+        return requests_queryset
+
+
+class SpeakingPracticeAttemptReviewCreateView(TeacherCourseViewMixin, CreateView):
+    model = SpeakingPracticeAttemptReview
+    template_name = "ComSemApp/teacher/attempt_request_info.html"
+    fields = ["request", "reviewer", "comments"]
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        attempt_id = self.kwargs.get('attempt_id')
+        attempt = get_object_or_404(SpeakingPracticeAttempt, id=attempt_id)
+        context['attempt'] = attempt
+        return context
+    
+
+    def form_invalid(self, form):
+        response = super().form_invalid(form)
+        return JsonResponse(form.errors, status=400)
+
+    def form_valid(self, form):
+        review = form.save(commit=False)
+        newScore = self.request.POST.get('newScore')
+        id = self.request.POST.get('request')
+        attempt = get_object_or_404(SpeakingPracticeAttempt, id=id)
+
+        if newScore is not None or newScore is not "":
+            attempt.correct = newScore
+            attempt.save()
+            review.original_score = self.request.POST.get('originalScore')
+        review.save()
+
+        return JsonResponse({}, status=200)
+    
+class SpeakingPracticeAttemptReviewUpdateView(TeacherCourseViewMixin, UpdateView):
+    model = SpeakingPracticeAttemptReview
+    template_name = "ComSemApp/teacher/attempt_request_info.html"
+    fields = ["reviewer", "comments"]
+    context_object_name = 'review'
+
+    def get_object(self, **kwargs):
+        attempt_id = self.kwargs.get('attempt_id')
+        review = get_object_or_404(SpeakingPracticeAttemptReview, request=attempt_id)
+        return review
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        attempt_id = self.kwargs.get('attempt_id')
+        attempt = get_object_or_404(SpeakingPracticeAttempt, id=attempt_id)
+        context['attempt'] = attempt
+        return context
+
+    def form_invalid(self, form):
+        response = super().form_invalid(form)
+        print(form.errors)
+        return JsonResponse(form.errors, status=400)
+
+    def form_valid(self, form):
+        review = form.save(commit=False)
+        newScore = self.request.POST.get('newScore')
+        id = self.request.POST.get('request')
+        attempt = get_object_or_404(SpeakingPracticeAttempt, id=id)
+
+        if newScore is not None or newScore is not "":
+            attempt.correct = newScore
+            attempt.save()
+            review.original_score = self.request.POST.get('originalScore')
```

Comment:
Since this is updating an existing review, the review's `original_score` should only be set if it has not been already (else it's not original, just most recent).